### DISCUSSION TITLE
yaw and induction control

### DISCUSSION THREAD

**Discussion by longseyilang** [NatLabRockies/floris#1043] [Q&A] [answered]
Can Floris model allow users to develop yaw and induction control  simultaneously to maximize wind farm’s power output?

**Comment by misi9170** [comment, marked as answer]
Hi @longseyilang , the [`MixedOperationTurbine` model](https://nrel.github.io/floris/operation_models_user.html#mixed-operation-model) allows any given turbine to be either derated (i.e., it's axial induction is reduced) OR yawed. Over a farm, some turbines may be derated while others are yaw misaligned. I don't think we currently have a direct example of this, but you could look at [this example](https://nrel.github.io/floris/examples/examples_control_types/003_setting_yaw_and_disabling.html) (also available [here](https://github.com/NREL/floris/blob/fd37e9b68fefe766cc5b989f8a8f40602913170e/examples/examples_control_types/003_setting_yaw_and_disabling.py)) and considering providing `power_setpoints` to the unyawed turbines instead of disabling them.

**Reply by paulf81** [reply]
Also want to flag that #924 and #832 will augment this capability with improved actuator disk models and models of turbines in yaw and non-nominal operating points

**Reply by longseyilang** [reply]
Thank you very much for this valuable feedback

**Reply by longseyilang** [reply]
Can the axial induction factor  be directly chosen as the control variable of the turbine in Floris model instead of derated power?

**Reply by zyla123** [reply]
I have also encountered this type of problem currently, perhaps we can communicate about it

**Reply by longseyilang** [reply]
Great. How to contact with you?

**Reply by paulf81** [reply]
Could you use the derating control, as in https://github.com/NREL/floris/blob/main/examples/examples_control_types/001_derating_control.py, to achieve the axial induction you want?

**Reply by longseyilang** [reply]
Thank you for your response! My aim is to improve wind farm power output using the axial induction control.  Using derating control based on power setpoint is not a good choice for me.  Do you have a plan to add a interface that  axial induction factor as control variable (similar to yaw angle) in Floris?

**Reply by paulf81** [reply]
Right now derating is the primary means of controlling axial induction.  However new pull requests will provide additional actuator models, for example #924 and these should provide additional means of controlling axial induction more exactly when merged.

**Reply by longseyilang** [reply]
Thank you very much!

**Comment by zyla123** [comment]
You can use this email address


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NREL/floris"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月12日(星期四) 下午2:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NREL/floris] yaw and induction control (Discussion #1043)





 
Great. How to contact with you?

**Reply by longseyilang** [reply]
According to my understanding, there is no interface for axial induction factor as control variable in Floris.

**Comment by zyla123** [comment]
I also want to solve this problem, and I haven't found any interface. How are you planning to solve it?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "NREL/floris"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月12日(星期四) 下午3:57
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [NREL/floris] yaw and induction control (Discussion #1043)





 
According to my understanding, there is no interface for axial induction factor as control variable in Floris.

**Reply by longseyilang** [reply]
Thank you！I have no  good idea.
